model_analyzer/perf_analyzer/perf_analyzer.py

```diff
@@ -280,8 +280,8 @@ def _poll_perf_analyzer(self, process):
                 logger.info(
                     f'perf_analyzer used significant amount of CPU resources ({cpu_util}%), killing perf_analyzer'
                 )
-                self._output = process.stdout.read()
                 process.kill()
+                self._output, _ = process.communicate()
```

Comment:
Does this need to be before process.kill? I think process.kill might be blocked because it cannot right anything to stdout/stderr?

Reply:
kill will kill immediately, after which communicate will still work
communicate will block until the process is done, which is not what we want.

From the official docs:
![image](https://user-images.githubusercontent.com/50968584/176737481-94e68212-a983-4f29-b8a5-120dc218c29b.png)

Reply:
I see, thanks for the clarification.